### PR TITLE
PY-61: add room registry implementation.

### PR DESCRIPTION
**Pull request by WinterCitizen**
* Move the project to poetry
* Add ruff & mypy linting

### CODE REVIEW

File: back/interfaces/room_registry.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        """Get all stored rooms."""
+        raise NotImplementedError
+
+    def get_room(self: Self, room_name: str) -> RoomInterface | None:
+        """Get room by name."""
+        raise NotImplementedError
+
+    def add_room(self: Self, room: RoomInterface) -> None:
+        """Add room to registry."""
+        raise NotImplementedError
```
Comment (UlanDT) [inline back/interfaces/room_registry.py:20]:
```
def delete_room(self: Self, room: RoomInterface) -> None:
    """Delete room from registry."""
    raise NotImplementedError
```

**Review by UlanDT — APPROVED** [review]
we should be able to delete room in case if all members exited room

**Review by aalysher — APPROVED** [review]
LGTM

File: back/room_registry.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    def add_room(self: Self, room: RoomInterface) -> None:
+        """Add room to registry."""
+        room_name = room.get_name()
+
+        with self.rooms_lock:
+            if room_name in self.rooms:
+                msg = f"Room {room_name} already exists"
+                raise ValueError(msg)
+
+            self.rooms[room_name] = room
```
[the remark this answers is not in the document]
Comment (WinterCitizen) [reply back/room_registry.py:34]: Done

File: back/interfaces/room_registry.py (hunk 1 above)
Comment (WinterCitizen) [reply back/interfaces/room_registry.py:20]: Done

**Review by UlanDT — APPROVED** [review]

**Review by aalysher — APPROVED** [review]
LGTM

**Pull request merged**